net-dns/coredns/coredns-1.7.0.ebuild

```diff
@@ -0,0 +1,887 @@
+	"gopkg.in/yaml.v2 v2.2.1/go.mod"
+	"gopkg.in/yaml.v2 v2.2.2/go.mod"
+	"gopkg.in/yaml.v2 v2.2.4/go.mod"
+	"gopkg.in/yaml.v2 v2.2.5/go.mod"
+	"gopkg.in/yaml.v2 v2.2.7/go.mod"
+	"gopkg.in/yaml.v2 v2.2.8/go.mod"
+	"honnef.co/go/tools v0.0.0-20180728063816-88497007e858/go.mod"
+	"honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod"
+	"honnef.co/go/tools v0.0.0-20190106161140-3f1c8253044a/go.mod"
+	"honnef.co/go/tools v0.0.0-20190418001031-e561f6794a2a/go.mod"
+	"honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod"
+	"honnef.co/go/tools v0.0.1-2019.2.3/go.mod"
+	"honnef.co/go/tools v0.0.1-2020.1.3/go.mod"
+	"k8s.io/api v0.18.3/go.mod"
+	"k8s.io/apimachinery v0.18.3/go.mod"
+	"k8s.io/client-go v0.18.3/go.mod"
+	"k8s.io/gengo v0.0.0-20190128074634-0689ccc1d7d6/go.mod"
+	"k8s.io/klog v0.0.0-20181102134211-b9b56d5dfc92/go.mod"
+	"k8s.io/klog v0.3.0/go.mod"
+	"k8s.io/klog v1.0.0/go.mod"
+	"k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6/go.mod"
+	"k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89/go.mod"
+	"rsc.io/binaryregexp v0.2.0/go.mod"
+	"rsc.io/quote/v3 v3.1.0/go.mod"
+	"rsc.io/sampler v1.3.0/go.mod"
+	"sigs.k8s.io/structured-merge-diff/v3 v3.0.0-20200116222232-67a7b8c61874/go.mod"
+	"sigs.k8s.io/structured-merge-diff/v3 v3.0.0/go.mod"
+	"sigs.k8s.io/yaml v1.1.0/go.mod"
+	"sigs.k8s.io/yaml v1.2.0/go.mod"
+)
+
+EGO_PN="github.com/${PN}/${PN}"
+
+inherit go-module
+go-module_set_globals
+
+ARCHIVE_URI="https://${EGO_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI="${ARCHIVE_URI} ${EGO_SUM_SRC_URI}"
+
+KEYWORDS="~amd64"
+
+DESCRIPTION="A DNS server that chains middleware"
+HOMEPAGE="https://github.com/coredns/coredns"
+
+LICENSE="Apache-2.0 MIT BSD ISC MPL-2.0 BSD-2"
+SLOT="0"
+
+src_compile() {
+	go build -v -ldflags="-X github.com/coredns/coredns/coremain.GitCommit=${EGIT_COMMIT}" || \
+		die "go build failed"
+}
+
+src_install() {
+	dobin "${PN}"
+	einstalldocs
+	doman man/*
+
+	newinitd "${FILESDIR}"/coredns.initd coredns
+	newconfd "${FILESDIR}"/coredns.confd coredns
+
+	insinto /etc/coredns/
+	newins "${FILESDIR}"/Corefile.example Corefile
+
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/coredns.logrotated coredns
+}
+
+src_test() {
+	# eclass default '-x' makes tests output unreadable
+	export GOFLAGS="-v -mod=readonly"
+
+	local known_fail=(
+		"TestZoneExternalCNAMELookupWithProxy"
+		"TestMetricsSeveralBlocs"
+		"TestMetricsAvailable"
+		"TestMetricsAvailableAfterReload"
+		"TestMetricsAvailableAfterReloadAndFailedReload"
+	)
+	# concat as '|^Test1$|^Test2$|^Test3...$':
+	local known_fail_re="$(printf '|^%s$' "${known_fail[@]}")"
+	# drop '|' in the begining:
+	known_fail_re="${known_fail_re:1}"
+
+	local working_tests_re="$(
+		# get list of all test:
+		{ GOFLLAGS="-mod=readonly" go test -list . ./... ||
+			die "Can't get list of tests"; } |
+		# skip "no tests" messages as well as know failures:
+		grep -v -E " |${known_fail_re}" |
+		# format a regexp:
+		sed -z 's/\n/$|^/g'
```

Comment:
... but since these happen in `src_test` which I find quite optional here, I don't mind. Just try to update these for the next version bump.